cmake/AwsCFlags.cmake

```diff
@@ -0,0 +1,69 @@
+# Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+include(CheckCCompilerFlag)
+
+# This function will set all common flags on a target
+# Options:
+#  NO_WGNU: Disable -Wgnu
+#  NO_WEXTRA: Disable -Wextra
+#  NO_PEDANTIC: Disable -pedantic
+function(aws_set_common_properties target)
+    set(options NO_WGNU NO_WEXTRA NO_PEDANTIC)
+    cmake_parse_arguments(SET_PROPERTIES "${options}" "" "" ${ARGN})
+
+    if(MSVC)
+        list(APPEND AWS_C_FLAGS /W4 /WX)
+    else()
+        list(APPEND AWS_C_FLAGS -Wall -Werror)
+
+        if(NOT SET_PROPERTIES_NO_WEXTRA)
+            list(APPEND AWS_C_FLAGS -Wextra)
+        endif()
+
+        if(NOT SET_PROPERTIES_NO_PEDANTIC)
+            list(APPEND AWS_C_FLAGS -pedantic)
+        endif()
+
+        # Warning disables always go last to avoid future flags re-enabling them
+        list(APPEND AWS_C_FLAGS -Wno-long-long)
+    endif()
+
+    if(NOT SET_PROPERTIES_NO_WGNU)
+        check_c_compiler_flag(-Wgnu HAS_WGNU)
+        if(HAS_WGNU)
+            # -Wgnu-zero-variadic-macro-arguments results in a lot of false positives
+            list(APPEND AWS_C_FLAGS -Wgnu -Wno-gnu-zero-variadic-macro-arguments)
+        endif()
+    endif()
+
+    if(CMAKE_BUILD_TYPE STREQUAL "" OR CMAKE_BUILD_TYPE MATCHES Debug)
+        list(APPEND AWS_C_DEFINES -DDEBUG_BUILD)
+    endif()
+
+    if(BUILD_SHARED_LIBS AND WIN32)
+        set(EXPORT_DEFINE ${target})
+        string(TOUPPER ${EXPORT_DEFINE} EXPORT_DEFINE)
+        string(REPLACE "AWS-C" "AWS" EXPORT_DEFINE ${EXPORT_DEFINE})
```

Comment:
This replace could false-positive and do something like this:
AWS-COMPRESSION" -> "AWSOMPRESSION"

maybe do the replace like:
string(REPLACE "^AWS-C-" "AWS-" ...